codalab/lib/upload_manager.py

```diff
@@ -258,3 +326,168 @@ def cleanup_existing_contents(self, bundle):
         bundle_update = {'data_hash': None, 'metadata': {'data_size': 0}}
         self._bundle_model.update_bundle(bundle, bundle_update)
         self._bundle_model.update_user_disk_used(bundle.owner_id)
+
+    def get_bundle_sas_token(self, path, **kwargs):
+        """
+        Get SAS token with write permission. Used for bypass server uploading.
+        """
+        return (
+            parse_linked_bundle_url(path)
+            .bundle_path_sas_url(permission='rw', **kwargs)
+            .split('?')[-1]
```

Comment:
What is `.split('?')[-1]` for? Can you add a comment?

Reply:
This is to get the SAS token from the SAS URL. I have add a comment for this, Thanks Ashwin!